File: server/src/lib/paths.py
from functools import lru_cache
from pathlib import Path

from src.env import get_settings


@lru_cache
def get_data_dir() -> Path:
    return Path(get_settings().data_dir)


@lru_cache
def get_seeds_dir() -> Path:
    return Path(get_settings().seeds_dir)


def skills_dir() -> Path:
    return get_data_dir() / "skills"


def incident_history_dir() -> Path:
    return get_data_dir() / "incident_history"


def knowledge_dir(project_slug: str | None = None) -> Path:
    base = get_data_dir() / "knowledge"
    return base / project_slug if project_slug else base


def seeds_skills_dir() -> Path:
    return get_seeds_dir() / "skills"

File: server/src/lib/paths.ts
import { join } from "path";
import { env } from "@/env";

export const dataDir = () => env.DATA_DIR;
export const seedsDir = () => env.SEEDS_DIR;
export const skillsDir = () => join(dataDir(), "skills");
export const uploadsDir = () => join(dataDir(), "uploads");
export const knowledgeDir = (projectSlug?: string) => {
  const base = join(dataDir(), "knowledge");
  return projectSlug ? join(base, projectSlug) : base;
};
export const seedsSkillsDir = () => join(seedsDir(), "skills");
